### DISCUSSION TITLE
Charge and discharge battery depending on forcast

### DISCUSSION THREAD

**Discussion by kristianolofsson** [davidusb-geek/emhass#381] [General]
Im currently depending on an other service that I would like to evaluate against this.

My current service works like this that every time an ask the service the json replay is providing a value by "Cheap", "Normal" and "Expensive". This return value is based on forecast energy price and possible feed from battery.

When Cheap I charge, when Normal I hold -> No charge/Discharge and only from grid. When expensive I use battery.

Eg. if the Price is high, but will get even higher the result might be "Normal" or even  to charge depending on how long the high price will last.

Im now interested in changing to this module. Will it serve me with the same result? And if yes, where can I get hold of state if to charge, hold or discharge depending on the prediction.

Currently using the a Huawei Luna solar system,

**Comment by JohanAlvedal** [comment]
It should go well, I have the same setup and run like that, although I have had to struggle with how to get started. Check the thread on the home assistant forum and there are lots of tips and examples of how to proceed..

**Comment by anselmhuewe** [comment]
are interested in checking out www.shyft-power.com (currently only in German). It manages battery charging/discharging based on an optimization modell which incorporates dynamic power tariffs.
There is no technical knowledge needed for the user to integrate.
